mlflow/tensorflow/__init__.py

```diff
@@ -709,7 +748,9 @@ def _load_pyfunc(path):
         should_compile = save_format == "tf"
         K = importlib.import_module(keras_module.__name__ + ".backend")
         if K.backend() == "tensorflow":
-            K.set_learning_phase(0)
+            # This API doesn't exist for tf > 2.15.0
+            if Version(tensorflow.__version__) <= Version("2.15.0"):
+                K.set_learning_phase(0)
```

Comment:
I don't think we need this line actually, this is supposed to be a deprecated API from ~2 years ago. We should be able to just delete this line, as the default behavior is inference mode.